### DISCUSSION TITLE
BlendeBot 2.0 and Internet Search

### DISCUSSION THREAD

**Discussion by jh-skaion** [facebookresearch/ParlAI#3985] [Q&A]
Hello team, 
i have been experimenting with blenderbot2.0 and internet search and i am trying to understand how the text generation works.
for example i asked the question: "i am doing fine. what is the latest with the terrorist attack at the airport in Kabul?"
it went through online searches where i can see it gather up to date information like:
![image](https://user-images.githubusercontent.com/82824999/131158782-af17bdc2-9994-4349-8226-8cafc657cd91.png)

but the response was: "I'm not sure, I haven't heard anything about it. Do you have a source for that?"

can anyone from the team tell how it works or point me to some documentation i can use to understand this process?

thank you
Juan

**Comment by jh-skaion** [comment]
This is another example:

![image](https://user-images.githubusercontent.com/82824999/131165822-56ed2259-a376-4ea4-90af-895f348bed6e.png)

**Comment by jh-skaion** [comment]
@stephenroller @klshuster

**Comment by klshuster** [comment]
The model attends to documents using the Fusion-in-Decoder method (FiD); see the [original paper](https://arxiv.org/abs/2007.01282) for more details of how the model incorporates external documents. 

Admittedly, sometimes the model does not attend to the documents, for a variety of reasons. Your second example, however, shows some attention (4th highest mountain is part of the retrieved knowledge).

**Reply by jh-skaion** [reply]
thanks for the response. i will read the paper to have a better understanding.

is there a search engine you can recommend?

**Reply by klshuster** [reply]
if you search around the ParlAI issues there are several suggested search engines
